client/rest-high-level/qa/ssl-enabled/build.gradle

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import org.elasticsearch.gradle.internal.test.RestIntegTestTask
+import org.elasticsearch.gradle.internal.info.BuildParams
+
+apply plugin: 'elasticsearch.java-rest-test'
+dependencies {
+  javaRestTestImplementation(testArtifact(project(':client:rest-high-level')))
+}
+
+tasks.matching{ it.name == "javaRestTest" }.configureEach {
+  systemProperty 'tests.rest.cluster.username', System.getProperty('tests.rest.cluster.username', 'test_user')
+  systemProperty 'tests.rest.cluster.password', System.getProperty('tests.rest.cluster.password', 'test-user-password')
+}
+
+testClusters.matching { it.name == 'javaRestTest' }.configureEach {
+  testDistribution = 'DEFAULT'
+  numberOfNodes = 2
+  systemProperty 'es.scripting.update.ctx_in_params', 'false'
+  setting 'reindex.remote.whitelist', '[ "[::1]:*", "127.0.0.1:*" ]'
+  setting 'xpack.license.self_generated.type', 'trial'
+  setting 'xpack.security.enabled', 'true'
+  setting 'xpack.security.authc.token.enabled', 'true'
+  setting 'xpack.security.authc.api_key.enabled', 'true'
+
+  extraConfigFile 'httpCa.p12', file('./src/javaRestTest/resources/httpCa.p12')
```

Comment:
Can we pick these up from the same place ?

Reply:
Rest Client has a method to read from resources of the class (see `findTrustStore()`), so I put httpCa there for client to find. I could have another copy of httpCa under `qa`, but it would create a potential problem if one copy is replaced in future. Alternatively, I can put transport.p12 under resources and read both files from there - not sure it would look better

Reply:
> Rest Client has a method to read from resources of the class (see findTrustStore()), so I put httpCa there for client to find.

findTruststore is not a RestClient method though right? We add it here in this test, and it basically just returns a `Path`, which we then use in the settings that RestClient is built with. In that sense we can put the file anywhere and as long as our findTruststore returns the Path properly, the RestClient will read this from the settings object. 

>  I can put transport.p12 under resources and read both files from there

That was my suggestion too.

Keeping both files at the same place ensures that we don't miss changing one of them if need arises, etc. Optimally, we'd also have a README.asciidoc in the same dir as the files, to describe how these resources are created if we need to recreate or change them in the future

Reply:
> Optimally, we'd also have a README.asciidoc in the same dir as the files, to describe how these resources are created if we need to recreate or change them in the future

Just repeating this in case you missed it.